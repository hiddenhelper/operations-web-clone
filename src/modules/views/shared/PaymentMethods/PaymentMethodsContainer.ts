import { IRootState } from '../../../state-mgmt/rootState';
import { connect } from 'react-redux';
import PaymentMethods from './PaymentMethods';
import { GENERAL } from '../../../../constants';
import { paymentState } from '../../../state-mgmt/payment';
import { generalState } from '../../../state-mgmt/general';

export const mapStateToProps = (state: IRootState) => ({
  paymentMethods: state.payment.paymentMethod,
  createLoading: state.general.loadingMap[GENERAL.LOADING_KEY.CREATE_PAYMENT],
  deleteLoading: state.general.loadingMap[GENERAL.LOADING_KEY.DELETE_PAYMENT_METHOD],
  fetchPaymentLoading: state.general.loadingMap[GENERAL.LOADING_KEY.FETCH_PAYMENT_METHODS],
});

export const mapDispatchToProps = dispatch => ({
  fetchPaymentMethods: () => dispatch(paymentState.actions.fetchPaymentMethodsStart()),
  deletePaymentMethod: (id: string) => dispatch(paymentState.actions.deletePaymentStart(id)),
  replaceAndDeletePaymentMethod: (toDeleteId: string, toReplaceWithId: string) =>
    dispatch(paymentState.actions.replaceAndDeletePaymentStart(toDeleteId, toReplaceWithId)),
  clearLoading: (key: string) => dispatch(generalState.actions.clear(key)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PaymentMethods);
