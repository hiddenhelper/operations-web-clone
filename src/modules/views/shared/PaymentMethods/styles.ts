import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants';
import { toREM } from '../../../../utils/generalUtils';

export const useStylesModalCards = makeStyles(theme =>
  createStyles({
    modalCards: {
      marginTop: 27,
      marginBottom: 20,
      justifyContent: 'center',
      paddingRight: 9,
      paddingLeft: 11,
      paddingTop: 0,
    },
    cardItemModal: {
      height: 170,
    },
    selectedCards: {
      margin: '47px 0px',
      justifyContent: 'center',
    },
    adminCards: {
      margin: '0px 0px',
      justifyContent: 'center',
    },
    note: {
      // marginTop: 42,
      width: '80%',
      margin: 'auto',
    },
    replaceModal: {
      width: 480,
      paddingBottom: 18,
    },
    deleteModalConfirmContent: {
      marginLeft: 0,
    },
    deleteModalText: {
      fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
      color: STYLE.COLOR.SEPTENARY,
      marginBottom: `${toREM(42)}`,
    },
    replaceModalButton: {
      minWidth: 180,
    },
    replaceModalText: {
      fontSize: `${toREM(STYLE.FONT.SIZE.PARAGRAPH)}`,
      color: STYLE.COLOR.SEPTENARY,
      marginBottom: `${toREM(15)}`,
    },
    deleteModalButton: {
      width: `${toREM(120)} !important`,
      marginLeft: 0,
    },
  })
);
