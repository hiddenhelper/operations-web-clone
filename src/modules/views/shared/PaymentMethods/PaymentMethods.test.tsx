import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PaymentMethods, { IPaymentMethodsProps } from './PaymentMethods';
import { getDefaultLoading, getPaymentMethodInUseResponse, getPaymentMethod_1, getPaymentMethod_2, getPaymentMethod_3 } from '../../../../test/entities';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { getInitialState } from '../../../../test/rootState';
import { MemoryRouter } from 'react-router';

describe('PaymentMethods', () => {
  let props: IPaymentMethodsProps;

  beforeEach(() => {
    props = {
      compact: false,
      fetchPaymentLoading: getDefaultLoading(),
      paymentMethods: [getPaymentMethod_1()],
      selectedPaymentMethod: getPaymentMethod_1().paymentMethodId,
      createLoading: getDefaultLoading(),
      deleteLoading: getDefaultLoading(),
      setSelectedPaymentMethod: jest.fn(),
      fetchPaymentMethods: jest.fn(),
      deletePaymentMethod: jest.fn(),
      replaceAndDeletePaymentMethod: jest.fn(),
      clearLoading: jest.fn(),
    };
  });

  it('should fetch payment methods', () => {
    render(<PaymentMethods {...props} />);

    expect(props.fetchPaymentMethods).toHaveBeenCalled();
  });

  it('should show loading indicator for admin', () => {
    props.admin = true;
    props.fetchPaymentLoading.isLoading = true;
    render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    expect(props.fetchPaymentMethods).toHaveBeenCalled();
  });

  it('should render with specific classes for project details', () => {
    props.isProjectDetail = true;
    render(<PaymentMethods {...props} />);

    expect(props.fetchPaymentMethods).toHaveBeenCalled();
  });

  it('should render admin mode', async () => {
    props.admin = true;
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.container).toMatchSnapshot();
  });

  it('should not let the user delete the last card', async () => {
    props.admin = true;
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    const deleteBtn = await wrapper.findByTestId('credit-card-item-delete');
    fireEvent.click(deleteBtn);
    const text = await wrapper.getByText(`This Credit Card can't be deleted`);
    expect(text).toBeDefined();
    const confirmBtn = await wrapper.findByTestId('confirm-button');
    fireEvent.click(confirmBtn);
  });

  it('should let the user delete a credit card', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_1(), getPaymentMethod_2(), getPaymentMethod_3()];
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    const deleteBtn = await wrapper.getAllByTestId('credit-card-item-delete')[0];
    await fireEvent.click(deleteBtn);
    const confirmBtn = await wrapper.findByTestId('confirm-button');
    await fireEvent.click(confirmBtn);

    expect(await props.deletePaymentMethod).toHaveBeenLastCalledWith(getPaymentMethod_1().paymentMethodId);
  });

  it('should render admin mode with the modal to resolve conflict', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_1(), getPaymentMethod_2(), getPaymentMethod_3()];
    props.deleteLoading.error = getPaymentMethodInUseResponse();

    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );
    const text = await wrapper.getByText(`Delete Credit Card **** **** **** ${getPaymentMethod_1().lastFourDigits}?`);
    expect(text).toBeDefined();
    const selectCard = await wrapper.findByTestId('replace-credit-card-select');
    await fireEvent.change(selectCard, { target: { name: 'replace-credit-card-select', value: 'e043ed03-1008-492a-ac40-4736713ee77f' } });

    const confirmBtn = await wrapper.findByTestId('modal-confirm-btn');
    fireEvent.click(confirmBtn);
    expect(props.replaceAndDeletePaymentMethod).toHaveBeenCalledWith(getPaymentMethod_1().paymentMethodId, getPaymentMethod_2().paymentMethodId);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render admin mode with the modal and do not delete if user close the modal', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_1(), getPaymentMethod_2(), getPaymentMethod_3()];
    props.deleteLoading.error = getPaymentMethodInUseResponse();
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );
    const closeBtn = await wrapper.findByTestId('modal-close-btn');
    fireEvent.click(closeBtn);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render admin mode and no open the modal if the payment with error does not exist', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_2(), getPaymentMethod_3()];
    props.deleteLoading.error = getPaymentMethodInUseResponse();
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render the disclamer modal and let the user to close the modal', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_1(), getPaymentMethod_2(), getPaymentMethod_3()];
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    const deleteBtn = await wrapper.getAllByTestId('credit-card-item-delete')[0];
    await fireEvent.click(deleteBtn);
    const confirmBtn = await wrapper.findByTestId('cancel-button');
    await fireEvent.click(confirmBtn);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should display a card replacement if the payment method is already used in a project', async () => {
    props.admin = true;
    props.paymentMethods = [getPaymentMethod_1(), getPaymentMethod_2(), getPaymentMethod_3()];
    const wrapper = await render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <PaymentMethods {...props} />
        </MemoryRouter>
      </Provider>
    );

    const deleteBtn = await wrapper.getAllByTestId('credit-card-item-delete')[1];
    await fireEvent.click(deleteBtn);
    const text = await wrapper.getByText(`Delete Credit Card **** **** **** ${getPaymentMethod_2().lastFourDigits}?`);
    expect(text).toBeDefined();
  });
});
