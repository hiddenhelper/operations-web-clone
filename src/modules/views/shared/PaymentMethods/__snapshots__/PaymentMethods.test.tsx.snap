// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethods should render admin mode 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-adminCards-160 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-194 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-204 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-213"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-206 MuiTypography-body1"
          >
            **** **** **** 4312
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-208"
          >
            <div
              class="makeStyles-creditCardLogo-207"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-203"
                src="card-logo-visa.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-205 MuiTypography-body1"
          >
            Rodolfo Perez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-205 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-202 makeStyles-cardItemClickable-195"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="makeStyles-emptyCardItem-193 false"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-215 MuiButton-textPrimary"
          data-testid="add-card-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg>
              credit-card.svg
            </svg>
            ADD NEW CREDIT CARD
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaymentMethods should render admin mode and no open the modal if the payment with error does not exist 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-adminCards-552 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-586 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-596 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-605"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-598 MuiTypography-body1"
          >
            **** **** **** 6213
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-600"
          >
            <div
              class="makeStyles-creditCardLogo-599"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-595"
                src="card-logo-mastercard.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-597 MuiTypography-body1"
          >
            Lucas Marquez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-597 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-594 makeStyles-cardItemClickable-587"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-586 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-596 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-605"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-598 MuiTypography-body1"
          >
            **** **** **** 7468
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-600"
          >
            <div
              class="makeStyles-creditCardLogo-599"
            >
              <svg
                class="makeStyles-creditCardIconImage-595"
              >
                credit-card.svg
              </svg>
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-597 MuiTypography-body1"
          >
            Nico  Monetto
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-597 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-594 makeStyles-cardItemClickable-587"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="makeStyles-emptyCardItem-585 false"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-607 MuiButton-textPrimary"
          data-testid="add-card-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg>
              credit-card.svg
            </svg>
            ADD NEW CREDIT CARD
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaymentMethods should render admin mode with the modal and do not delete if user close the modal 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiGrid-root makeStyles-adminCards-446 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-480 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-490 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-499"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-492 MuiTypography-body1"
          >
            **** **** **** 4312
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-494"
          >
            <div
              class="makeStyles-creditCardLogo-493"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-489"
                src="card-logo-visa.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            Rodolfo Perez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-488 makeStyles-cardItemClickable-481"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-480 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-490 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-499"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-492 MuiTypography-body1"
          >
            **** **** **** 6213
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-494"
          >
            <div
              class="makeStyles-creditCardLogo-493"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-489"
                src="card-logo-mastercard.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            Lucas Marquez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-488 makeStyles-cardItemClickable-481"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-480 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-490 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-499"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-492 MuiTypography-body1"
          >
            **** **** **** 7468
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-494"
          >
            <div
              class="makeStyles-creditCardLogo-493"
            >
              <svg
                class="makeStyles-creditCardIconImage-489"
              >
                credit-card.svg
              </svg>
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            Nico  Monetto
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-491 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-488 makeStyles-cardItemClickable-481"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="makeStyles-emptyCardItem-479 false"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-501 MuiButton-textPrimary"
          data-testid="add-card-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg>
              credit-card.svg
            </svg>
            ADD NEW CREDIT CARD
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaymentMethods should render admin mode with the modal to resolve conflict 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiGrid-root makeStyles-adminCards-340 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-374 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-384 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-393"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-386 MuiTypography-body1"
          >
            **** **** **** 4312
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-388"
          >
            <div
              class="makeStyles-creditCardLogo-387"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-383"
                src="card-logo-visa.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            Rodolfo Perez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-382 makeStyles-cardItemClickable-375"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-374 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-384 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-393"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-386 MuiTypography-body1"
          >
            **** **** **** 6213
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-388"
          >
            <div
              class="makeStyles-creditCardLogo-387"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-383"
                src="card-logo-mastercard.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            Lucas Marquez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-382 makeStyles-cardItemClickable-375"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-374 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-384 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-393"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-386 MuiTypography-body1"
          >
            **** **** **** 7468
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-388"
          >
            <div
              class="makeStyles-creditCardLogo-387"
            >
              <svg
                class="makeStyles-creditCardIconImage-383"
              >
                credit-card.svg
              </svg>
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            Nico  Monetto
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-385 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-382 makeStyles-cardItemClickable-375"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="makeStyles-emptyCardItem-373 false"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-395 MuiButton-textPrimary"
          data-testid="add-card-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg>
              credit-card.svg
            </svg>
            ADD NEW CREDIT CARD
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaymentMethods should render the disclamer modal and let the user to close the modal 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiGrid-root makeStyles-adminCards-612 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-646 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-656 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-665"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-658 MuiTypography-body1"
          >
            **** **** **** 4312
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-660"
          >
            <div
              class="makeStyles-creditCardLogo-659"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-655"
                src="card-logo-visa.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            Rodolfo Perez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-654 makeStyles-cardItemClickable-647"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-646 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-656 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-665"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-658 MuiTypography-body1"
          >
            **** **** **** 6213
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-660"
          >
            <div
              class="makeStyles-creditCardLogo-659"
            >
              <img
                alt="Credit Card Logo"
                class="makeStyles-creditCardIconImage-655"
                src="card-logo-mastercard.png"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            Lucas Marquez
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-654 makeStyles-cardItemClickable-647"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-cardItem-646 MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="card-item"
      >
        <p
          class="MuiTypography-root makeStyles-creditCardTitle-656 MuiTypography-body1"
        >
          CREDIT CARD
        </p>
        <div
          class="makeStyles-cardNumberWrapper-665"
        >
          <p
            class="MuiTypography-root makeStyles-creditCardNumber-658 MuiTypography-body1"
          >
            **** **** **** 7468
          </p>
          <div
            class="makeStyles-creditCardLogoContainer-660"
          >
            <div
              class="makeStyles-creditCardLogo-659"
            >
              <svg
                class="makeStyles-creditCardIconImage-655"
              >
                credit-card.svg
              </svg>
            </div>
          </div>
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            Nico  Monetto
          </p>
          <p
            class="MuiTypography-root makeStyles-creditCardHolder-657 MuiTypography-body1"
          >
            USA
          </p>
        </div>
        <span
          class="makeStyles-cardSelectedIcon-654 makeStyles-cardItemClickable-647"
          data-testid="credit-card-item-delete"
          role="button"
        >
          <svg
            style="cursor: pointer;"
            title="Delete Card"
          >
            trash-can.svg
          </svg>
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
    >
      <div
        class="makeStyles-emptyCardItem-645 false"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addNewCardButton-667 MuiButton-textPrimary"
          data-testid="add-card-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg>
              credit-card.svg
            </svg>
            ADD NEW CREDIT CARD
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
